targets/csharp/source/source/Uunit/PluginManagerTest.cs

```diff
@@ -0,0 +1,353 @@
+        /// Test that SDK can use a custom transport plugin.
+        /// </summary>
+        [UUnitTest]
+        public void SdkCustomPluginTransport(UUnitTestContext testContext)
+        {
+            var playFabTransport = PluginManager.GetPlugin(PluginContract.PlayFab_Transport) as ITransportPlugin;
+            var customTransport = new CustomTransportPlugin();
+
+            try
+            {
+                // Set a custom transport plugin
+                PluginManager.SetPlugin(customTransport, PluginContract.PlayFab_Transport);
+
+                // Call some PlayFab API 
+                var getRequest = new GetUserDataRequest();
+                var getDataTask = PlayFabClientAPI.GetUserDataAsync(getRequest, null, extraHeaders);
+                ContinueWithContext(getDataTask, testContext, null, false, "GetUserData call failed", false);
+                getDataTask.Wait();
+
+                // Verify
+                testContext.True(customTransport.IsDoPostCalled);
+                testContext.EndTest(UUnitFinishState.PASSED, null);
+            }
+            catch (Exception e)
+            {
+                testContext.EndTest(UUnitFinishState.FAILED, e.ToString());
+            }
+            finally
+            {
+                // Restore the original plugin
+                PluginManager.SetPlugin(playFabTransport, PluginContract.PlayFab_Transport);
+            }
+        }
+
+        /// <summary>
+        /// CLIENT AND SERVER API
+        /// Test that multiple plugins can be used with the same contract simultaneously.
+        /// </summary>
+        [UUnitTest]
+        public void SdkMultiplePluginsPerContract(UUnitTestContext testContext)
+        {
+            // Create multiple custom plugins, set them with the same contract
+            const int pluginNumber = 4;
+            var APIs = new CustomApi[pluginNumber];
+            for (int i = 0; i < pluginNumber; i++)
+            {
+                var pluginName = "Plugin_" + i;
+                var customTransport = new CustomTransportPlugin() { Name = pluginName };
+                PluginManager.SetPlugin(customTransport, PluginContract.PlayFab_Transport, pluginName);
+                APIs[i] = new CustomApi() { PluginName = pluginName };
+            }
+
+            try
+            {
+                // Call the APIs
+                int totalNumberOfCalls = 0;
+                var resultTasks = new ConcurrentBag<Task<object>>();
+                Parallel.ForEach(APIs, api =>
+                {
+                    // Each API uses its specific transport plugin
+                    Task<object> task = api.SomeApiMethod();
+                    resultTasks.Add(task);
+                    totalNumberOfCalls++;
+                });
+
+                var resultTasksArray = resultTasks.ToArray();
+                Task.WaitAll(resultTasksArray);
+
+                // Verify
+                testContext.True(totalNumberOfCalls == pluginNumber);
```

Comment:
This line doesn't actually test anything about your plugins.
You set totalNumberOfCalls totally independently of anything your plugins do, and thus the test condition doesn't actually contribute to what you're trying to test.